File: src/transformers/features/notations/NotationAssertGeneralTransformer.ts
import { NotationAssertGeneralProgrammer } from "../../../programmers/notations/NotationAssertGeneralProgrammer";

import { StringUtil } from "../../../utils/StringUtil";

import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace NotationAssertGeneralTransformer {
  export const transform = (rename: (str: string) => string) =>
    GenericTransformer.scalar(
      `notations.assert${StringUtil.capitalize(rename.name)}`,
    )(
      (project) => (modulo) =>
        NotationAssertGeneralProgrammer.write(rename)(project)(modulo),
    );
}

File: packages/transform/src/features/notations/NotationAssertGeneralTransformer.go
package notations

import (
	corenotations "github.com/samchon/typia/packages/core/src/programmers/notations"
	internal "github.com/samchon/typia/packages/transform/src/internal"
)

var NotationAssertGeneralTransformer = notationAssertGeneralTransformerNamespace{}

type notationAssertGeneralTransformerNamespace struct{}

func (notationAssertGeneralTransformerNamespace) Transform(rename string) internal.Task {
	return func(call internal.ITypiaCallExpression) string {
		return corenotations.NotationAssertGeneralProgrammer.Write(rename, call.TypeText)
	}
}
